### PR TITLE
JRA-116 Adde pre-select destination for favorites

### PR DESCRIPTION
**Pull request by YawarValeriano**
Preselects destination for favorites
Modified logic for displaying destination from different views

### CODE REVIEW

File: RoutesApp_ios/modules/favoritesList/controllers/FavoritesListViewController.swift [hunk 1]
```diff
@@ -121,4 +121,15 @@ extension FavoritesListViewController: UITableViewDelegate, UITableViewDataSourc
...
     }
+
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        let fav = viewmodel.getFavoriteAt(index: indexPath.row)
+        let tabViewController = SceneDelegate.shared?.window?.rootViewController as? UITabBarController
+        guard let tabController = tabViewController else { return }
+        tabController.selectedIndex = 2
+        guard let navController = tabController.selectedViewController as? UINavigationController else {return}
+        guard let homeController = navController.viewControllers[0] as? HomeViewController else { return }
+        let coordinates = Coordinate(latitude: Double(fav.latitude!)!, longitude: Double(fav.longitude!)!)
```
Comment (jmacboy) [inline RoutesApp_ios/modules/favoritesList/controllers/FavoritesListViewController.swift]: do not force unwrap here please, validate before doing this

File: RoutesApp_ios/modules/home/controllers/HomeViewController.swift [hunk 2]
```diff
@@ -192,6 +201,20 @@ class HomeViewController: UIViewController {
         }
     }
 
+    private func gotToTab() {
+        destinationContainer.isHidden = true
+        let tabViewController = SceneDelegate.shared?.window?.rootViewController as? UITabBarController
+        guard let tabController = tabViewController else { return }
+        switch destinationFromDifferentController {
+        case .TOURPOINTS:
+            tabController.selectedIndex = 1
```
Comment (jmacboy) [inline RoutesApp_ios/modules/home/controllers/HomeViewController.swift]: use const values to indicate which tab is which on the selectedIndex

**Review by jmacboy — CHANGES_REQUESTED** [review]

File: RoutesApp_ios/modules/favoritesList/controllers/FavoritesListViewController.swift (hunk 1 above)
Comment (YawarValeriano) [reply RoutesApp_ios/modules/favoritesList/controllers/FavoritesListViewController.swift]: Fixed

File: RoutesApp_ios/modules/home/controllers/HomeViewController.swift (hunk 2 above)
Comment (YawarValeriano) [reply RoutesApp_ios/modules/home/controllers/HomeViewController.swift]: Changed for const values

**Review by jmacboy — APPROVED** [review]

**Pull request merged**